src/tdastro/astro_utils/dust_map.py

```diff
@@ -0,0 +1,141 @@
+
+        Returns
+        -------
+        dust_map : `dustmaps.DustMap`
+            A "query" object for the requested dustmap.
+        """
+        # Find the correct submodule within dustmaps and load it.
+        dm_module = None
+        for submodule in iter_modules(dustmaps.__path__):
+            if name == submodule.name:
+                dm_module = importlib.import_module(f"dustmaps.{name}")
+        if dm_module is None:
+            raise KeyError(f"Invalid dustmap '{name}'")
+
+        # Fetch the data to TDAstro's cache directory.
+        dm_config["data_dir"] = str(_TDASTRO_CACHE_DATA_DIR / "dustmaps")
+        dm_module.fetch()
+
+        # Get the query object by searching for a class using the {Module}Query
+        # naming convention.
+        target_name = f"{name}query"
+        query_class_name = None
+        for attr in dir(dm_module):
+            if attr.lower() == target_name:
+                query_class_name = attr
+        if query_class_name is None:
+            raise ValueError(f"Unable to find query class within module dustmaps.{name}")
+
+        # Get the class, create a query object, and return that object.
+        dm_class = getattr(dm_module, query_class_name)
+        return dm_class()
+
+    @staticmethod
+    def load_extinction_model(name, **kwargs):
+        """Load the extinction model.
+
+        Parameters
+        ----------
+        name : str
+            The name of the extinction model to use.
+        **kwargs : dict
+            Any additional keyword arguments needed to create that argument.
+
+        Returns
+        -------
+        ext_obj
+            A extinction object.
+        """
+        for submodule in iter_modules(dust_extinction.__path__):
+            ext_module = importlib.import_module(f"dust_extinction.{submodule.name}")
+            if ext_module is not None and name in dir(ext_module):
+                ext_class = getattr(ext_module, name)
+                return ext_class(**kwargs)
+        raise KeyError(f"Invalid dust extinction model '{name}'")
+
+    def apply(self, flux_density, wavelengths, ra, dec, dist=1.0):
+        """Apply the effect to observations (flux_density values)
+
+        Parameters
+        ----------
+        flux_density : `numpy.ndarray`
+            An array of flux density values (in nJy).
+        wavelengths : `numpy.ndarray`, optional
+            An array of wavelengths (in angstroms).
+        ra : `float`
+            The object's right ascension (in degrees).
+        dec : `float`
+            The object's declination (in degrees).
+        dist : `float`
+            The object's distance (in ?).
+            Default = 1.0
+
+        Returns
+        -------
+        flux_density : `numpy.ndarray`
+            The results (in nJy).
+        """
+        # Get the extinction value at the object's location.
+        coord = SkyCoord(ra, dec, dist, frame="icrs", unit="deg")
+        ebv = self.dust_map.query(coord)
```

Comment:
The problem with this approach is that different maps return different values - it is not always E(B-V). An example:
https://dustmaps.readthedocs.io/en/latest/modules.html#module-dustmaps.bayestar